src/avt/FileWriter/avtDatasetFileWriter.C

```diff
@@ -371,12 +371,20 @@ avtDatasetFileWriter::WriteOBJTree(avtDataTree_p dt, int idx,
 //
 //    Kathleen Biagas, Fri Feb 22 15:39:02 PST 2013
 //    If using cd2pd, use it's ouput port as input to the geometry filter.
+// 
+//    Justin Privitera, Fri Nov  3 15:25:32 PDT 2023
+//    The new arguments (writeMTL, MTLHasTex, and texFilename) are used to
+//    setup needed parameters for the upgraded vtkOBJWriter.
 //
 // ****************************************************************************
 
 void
-avtDatasetFileWriter::WriteOBJFile(vtkDataSet *ds, const char *fname,
-                                   const char *label)
+avtDatasetFileWriter::WriteOBJFile(vtkDataSet *ds,
+                                   const char *fname,
```

Comment:
Hmmm...that seems like a lot of OBJ specific stuff in what is intended to be a general (base) class for writers for databases.

If some stuff at this level is truly needed, I worry we're starting down a slope of the union of all possible file writer parameters from all possible databases showing up here as args to this method.

Is there any way to package and genericize this a bit...I mean, can we pass something like a `Node` (or some other jsony-like-thing) that is...

Hmmmm...wait a minute...it looks like there was A LOT of OBJ specific stuff here before your change for this PR. Ok, forget I said anything for now. But, I wanna capture this comment here for now too...just ignore it.

Reply:
Yeah, the `avtDatasetFileWriter` has very specific code for multiple different file types. It seems like lots of logic is reimplemented for plugins. I want to investigate that further in the future (#19031).